system_pre.js

```diff
@@ -0,0 +1,29 @@
+if (threading_supported) {
+  if (typeof SharedInt32Array === 'undefined') {
+    var makeSharedTypedArrayCtor = function(ctor) {
+      return function(arg) {
+        if (typeof arg === 'number') {
+          // Constructing a new SharedTypedArray of size arg.
+          var sab = new SharedArrayBuffer(arg * ctor.BYTES_PER_ELEMENT);
+          return new ctor(sab);
+        } else {
+          // Constructing an ArrayBufferView with a given buffer. Forward the
+          // args through.
+          var args = Array.prototype.slice.call(arguments);
+          args.unshift(null);
+          var bound_ctor = ctor.bind.apply(ctor, args);
+          return new bound_ctor();
+        }
+      };
+    };
+
+    SharedInt8Array = makeSharedTypedArrayCtor(Int8Array);
```

Comment:
Not sure I understand the question. Do you mean why can't I just say SharedInt32Array = Int32Array? If so, it is because the SharedInt32Array(<number>) constructor implicitly creates a SharedArrayBuffer.